app/views/contexts/section_summary_context.py

```diff
@@ -133,15 +141,18 @@ def _build_summary(self, return_to: Optional[str]):
 
         return groups
 
-    def _title_for_location(self):
+    def _title_for_location(self) -> Union[str, dict]:
         section_id = self.current_location.section_id
         return (
-            self._schema.get_repeating_title_for_section(section_id)
+            # Type ignore: section id should exist at this point
+            self._schema.get_repeating_title_for_section(section_id)  # type: ignore
             or self._schema.get_summary_title_for_section(section_id)
             or self._schema.get_title_for_section(section_id)
         )
 
-    def _custom_summary_elements(self, section_summary):
+    def _custom_summary_elements(
+        self, section_summary: Iterable[dict]
```

Comment:
Unless we are mutating or using dict specific method, this should be `Iterable[Mapping]`.

Reply:
We seems to be mutating it at `placeholder_render.py` line 107

Reply:
Thanks for the explation @MebinAbraham . Will update